scripts/fastapi/cache_operations.py

```
@@ -163,14 +164,31 @@ def complete_daily_in_cache(username: str, date: str) -> bool:
+        # Update user's 'today' field and calculate new streak
         cached_users = cache_manager.get(CacheType.USERS)
         if cached_users and cached_users.get('success'):
             users = cached_users.get('data', [])
             user = next((u for u in users if u.get('username') == username), None)
             if user:
                 user['today'] = 1
 
+                # Calculate and update streak
+                from datetime import datetime, timedelta
+                today_date = datetime.strptime(date, '%Y-%m-%d')
+                yesterday_date = (today_date - timedelta(days=1)).strftime('%Y-%m-%d')
```

Ruff (0.14.5)

Local variable `yesterday_date` is assigned to but never used

Remove assignment to unused variable `yesterday_date`

(F841)